core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -138,6 +137,7 @@ trait Runtime[+R] { self =>
           fiber.tell(
             FiberMessage.Stateful(fiber => fiber.addObserver(exit => result.set(exit.asInstanceOf[Exit[E, A]])))
           )
+          Executor.signalBlocking()
```

Comment:
doesn't it make sense to wrap the blocking call with a scope aware API (as suggested above)? this way capable executors will be able to reuse the current worker instead of stopping it and later spawning a new one

Reply:
The executor isn't stopping it per-se. It allows is to continue execution, but it removes the thread that is about to block from the _very_ bounded threadpool and add a new thread in it to it. Once this thread finishes execution, it'll be placed in a cache, and it can be reused again if we ever reach this point.

Unless you're running A LOT of `Runtime.unsafe.run` in parallel within another Fiber, it should never be an issue

Reply:
As for making it scoped, I'm not sure if it's possible. We're not marking the downstream effect as blocking, we're marking the upstream worker as blocking

Reply:
`ZScheduler` isn't the only executor implementation, other may have other means to handle blocking.
re `ZScheduler` itself, we don't want to modify the downstream effect, we just want to temporarily mark the current worker as blocking so the scheduler can take this into account when managing its resources, once we know the blocking is done we can inform the scheduler again so it can reconsider this worker as part of its general resource management. i.e. when the worker informs the scheduler it's becoming blocked the scheduler can decrease the number of active workers, not necessarily allocate a new one immediately, then when the worker informs the scheduler its no longer blocked the scheduler can return it into the active workers set, place it in a cache or terminate it.